Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.7.7-slim
+
+# Create virtualenv and add to path.
+ENV VIRTUAL_ENV=/opt/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+# System dependencies
+RUN apt-get update && apt-get install -y build-essential && rm -rf /var/lib/apt/lists/*
```

Comment:
Once these system dependencies are installed, should I create a non-privileged user and run everything else as that?

Reply:
no - the container is isolated from the rest of your system anyway.

Reply:
@dooferlad This isn't quite true, as the expected behaviour is to mount host files to the container.  Leaving the user as root in this case give the container root access to all host files in the mounted directory, as the root user id is always 1.

On the other hand, any user you create in this dockerfile cannot be assumed to match any existing user on the host, so this will mean the container will either have full root access or absolute minimum access to the mounted files.